haas/api.py

```diff
@@ -133,7 +133,60 @@ def project_detach_node(project, node):
     project.nodes.remove(node)
     db.session.commit()
 
+@rest_call('PUT', '/network/<network>/access/<project>')
+def network_grant_project_access(project, network):
+   """Add access to <network> to <project>.
+   If the project or network does not exist, a NotFoundError will be raised.
+   """
+   network = _must_find(model.Network, network)
+   project = _must_find(model.Project, project)
+   #must be admin or the owner of the network to add projects
+   get_auth_backend().require_project_access(network.owner)
+
+   if project in network.access:
+       raise DuplicateError('Network %s is already in project %s'%
+                            (network.label, project.label))
+
+   network.access.append(project)
+   db.session.commit()
+
+@rest_call('DELETE', '/network/<network>/access/<project>')
+def network_revoke_project_access(project, network):
+    """Remove access to <network> from <project>.
+    If the project or network does not exist, a NotFoundError will be raised.
+    If the project is the owner of the network a BlockedError will be raised.
+    """
+    network = _must_find(model.Network, network)
+    project = _must_find(model.Project, project)
+    #must be admin, the owner of the network, or <project> to remove <project>.
+    authorized = get_auth_backend().have_project_access(network.owner)
+    
+    if network.access:
+        for proj in network.access:
+            authorized = authorized or ((proj.label == project.label) and get_auth_backend().have_project_access(proj))
+            
+    if not authorized:
+        raise AuthorizationError("You do not have access to this project.")
+   
+    if project not in network.access:
+        raise NotFoundError("Network %s is not in project %s"%
```

Comment:
Suggest using `%r` instead of `%s`, which will quote the value (when it's a string anyway).